G2: UI surrounding the editor canvas

youknowriad opened the pull request:
bulids on top of #19344 with some additional items that can be harder to land without changing the whole canvas:

 - [x] Dropdowns with darker borders
 - [ ] Updated border-less panels
 - [ ] Darker borders for sidebar, header and footer.

youknowriad committed:
G3: UI surrounding editor canvas branch start.

- Updates borders for dropdowns, and proceeds to do additional polish work outside the editor canvas.

Commit:
General layout polish.

jasmussen commented:
I rebased this off G2 (also recently rebased), and started initial work.

But I'm starting slowly with this, because we are getting into some somewhat deep surgery of the editor — a reduction in colors (3 shades of gray instead of 10), a reduction, and other simplification work. For that reason in my first commit I focused only on the popovers in https://github.com/WordPress/gutenberg/pull/19348/commits/ba4fa80c750f313b9ab626c788bffce8f5779432 — yet that still touched a lot of files, so feel free to glance over them to get a flavor for some of the change work we're looking at.

And it's not even done, it's only this far:

<img width="316" alt="Screenshot 2020-01-22 at 12 56 37" src="https://user-images.githubusercontent.com/1204802/72892537-0d020600-3d17-11ea-9ec1-6ba3d76a69fc.png">

This needs padding and spacing alignments, the icons and keyboard shortcuts need to be on the right, and we need a special heuristic for the popover component: most of them should have the light gray border and shadow, but all of the popover menus inside the editing canvas should have the dark border and no shadow. 

Because there are always opinions on this, I don't want to move too fast.

jasmussen commented:
I took the liberty of renaming this PR. Can we rename the branch also? If it's tricky, no trouble.

youknowriad commented:
We could rename the branch but I think it would required creating a separate PR. So maybe just keep it as is.

ZebulanStanphill commented on packages/components/src/menu-group/style.scss:
```diff
@@ -1,15 +1,12 @@
 }
 
 .components-menu-group__label {
+	text-transform: uppercase;
```
Isn't all-caps considered worse for accessibility?

jasmussen replied:
I wasn't aware they were. 

I personally think that for a menu group label, which is decidedly seconary (light gray, smaller font size), it's okay to use it as a visual element, it helps the label also act as a separtor. However I'm open to suggestions, any thoughts @pablohoneyhoney?

jasmussen commented:
Made slight progress on simplifying the toolbar, which includes introducing a basic grid, rearranging the top left buttons, and moving the document outline to the footer (fixes #4287):

<img width="1034" alt="Screenshot 2020-01-24 at 12 04 41" src="https://user-images.githubusercontent.com/1204802/73064675-0eac0500-3ea2-11ea-94fd-2cbc46f05875.png">

**Please do not consider the above representative of anything other than work in progress.**

While the simplificaiton of the toolbar (grid, buttons), work is very far from done, and honestly this reduced border-style makes even clearer that we are in need of icons that fit together better. As the UI is reduced, it increasingly becomes clear that we are using a mix of icons and styles here. To no-ones fault, but a good reason for a fresh singular approach.

youknowriad commented on packages/block-editor/src/components/inserter/style.scss:
```diff
@@ -28,12 +28,24 @@ $block-inserter-search-height: 38px;
 	transition: color 0.2s ease;
+
+	// Spot color button.
+	// @todo: This button should ideally be 32px.
```
The inserter is used in different places and not just the top toolbar. I think the "blue button" shouldn't be solved using CSS but using the `isPrimary` prop in JS and only for the top toolbar inserter.

ZebulanStanphill commented:
One thing that has always confused me is that the <kbd><samp>Save Draft</samp></kbd> and <kbd><samp>Preview</samp></kbd> buttons use different styles. I would think that they should look the same.

Also, should the Document Outline button use a visible label? I would think that there is enough space for it. I'm also not convinced it belongs in the footer. Perhaps it should be an extension similar to what Yoast and other plugins have, with the ability to pin it to the right-side of the top toolbar.

Finally, I wonder if the Block Navigation menu belongs in the footer instead of the Document Outline. It has a similar function to the breadcrumb links. Or maybe they both belong there.

jasmussen commented:
> One thing that has always confused me is that the Save Draft and Preview buttons use different styles. I would think that they should look the same.

I still hope that we can entirely deprecate Save Draft as a vestige of the past. However it's increasingly clear that this is still a ways out in the future. In the mean time, yes, it does make sense to unify on a single button style. Though it is worth noting that there might land a different interface for the Preview button, when the editor [gets responsive features and registers an editor style](https://github.com/WordPress/gutenberg/pull/19082#issuecomment-567508590).

> Also, should the Document Outline button use a visible label? I would think that there is enough space for it. I'm also not convinced it belongs in the footer. Perhaps it should be an extension similar to what Yoast and other plugins have, with the ability to pin it to the right-side of the top toolbar.

IMO, it should be a word count visible label, yes. Like so:

<img width="1133" alt="Screenshot 2020-01-27 at 10 10 38" src="https://user-images.githubusercontent.com/1204802/73162074-478bea00-40ed-11ea-8f08-1cc510d2ab4d.png">

Then you click it, and the document outline interface shows up. The icon isn't particularly helpful there, but a label showing real time information combined with a tooltip feels like a good balance to me. 

> Finally, I wonder if the Block Navigation menu belongs in the footer instead of the Document Outline. It has a similar function to the breadcrumb links. Or maybe they both belong there.

Possibly. While I think the Selection mode and breadcrumb interfaces both have made strides in selecting child blocks, I don't think we'll know how efficient those two are until we actually ship the G2 efforts: specifically the changes to what has focus and what is selected, in my experience of working on the branches so far, make it much clearer what goes on.

That's a roundabout way of saying that until we feel confident enough that those two interfaces work for selecting parent blocks, perhaps we should keep the block navigation menu where it is? And if they prove themselves, your points do make sense that the button could serve as an extension/expansion of the breadcrumbs.

ZebulanStanphill commented:
> > Also, should the Document Outline button use a visible label? I would think that there is enough space for it. I'm also not convinced it belongs in the footer. Perhaps it should be an extension similar to what Yoast and other plugins have, with the ability to pin it to the right-side of the top toolbar.
> 
> IMO, it should be a word count visible label, yes. Like so:
> 
> <img alt="Screenshot 2020-01-27 at 10 10 38" width="1133" src="https://user-images.githubusercontent.com/1204802/73162074-478bea00-40ed-11ea-8f08-1cc510d2ab4d.png">
> 
> Then you click it, and the document outline interface shows up. The icon isn't particularly helpful there, but a label showing real time information combined with a tooltip feels like a good balance to me.

I agree that the icon isn't that helpful in the footer, but I don't think the word count makes for a good button to open up the Document Outline, since it is not a description of what the button does. We would have to use hidden labels on screen readers, and for everyone else it wouldn't be clear that word count is a button in the first place.

jasmussen commented:
> I don't think the word count makes for a good button to open up the Document Outline, since it is not a description of what the button does.

I actually quite disagree with this — the button shows a word count and shows word-count related information when clicked. 

I do agree it could use additional sanity checking, so I'll add a label and invite that.

jasmussen commented:
In this branch I refactored how hover and focus styles are specified to greatly reduce specificity. So far it's working, and as a result a ton of extra CSS can be removed and simplified. But it needs testing in all edgecases, so look for hover and focus to work in every situation.

Unified the add block buttons. It really deserves a new icon soon. 

<img width="695" alt="Screenshot 2020-02-06 at 12 25 24" src="https://user-images.githubusercontent.com/1204802/73933037-e4176e80-48db-11ea-8fb8-8074fddb2ea8.png">

ZebulanStanphill commented:
The empty Paragraph and placeholder inserters haven't changed. Is that intentional?
![image](https://user-images.githubusercontent.com/19592990/73962616-5f752200-48d4-11ea-907b-34928f443527.png)

Accordions still use the default focus style. Is that intentional?
![image](https://user-images.githubusercontent.com/19592990/73962678-787dd300-48d4-11ea-8603-c7b1d8aaf730.png)

Accordions also don't have blue text when you hover them. In fact, they don't have any hover styles at all.
![image](https://user-images.githubusercontent.com/19592990/73963368-c0512a00-48d5-11ea-9f08-495402964543.png)

The sidebar tabs don't use blue text on hover either.
![image](https://user-images.githubusercontent.com/19592990/73963947-e1fee100-48d6-11ea-8752-ee98fcaf4223.png)

There seems to be a bit too much space at the bottom of the <kbd><samp>More tools & options</samp></kbd> menu.
![image](https://user-images.githubusercontent.com/19592990/73963065-37d28980-48d5-11ea-9c60-fa397207ba34.png)

These buttons don't line up horizontally, even though you'd think they would. There's not much padding in the sidebar compared to the top bar.
![image](https://user-images.githubusercontent.com/19592990/73962996-14a7da00-48d5-11ea-94a3-d011748070c3.png)

<kbd><samp>Remove from Reusable blocks</samp></kbd> still has an icon.
![image](https://user-images.githubusercontent.com/19592990/73963551-1e7e0d00-48d6-11ea-986e-80e6f0b0a9d6.png)

The focus border is thicker for the select control than for other UI elements. Also, the text in these controls seems kind of small. Is that intentional?
![image](https://user-images.githubusercontent.com/19592990/73963836-a8c67100-48d6-11ea-9ff7-d78160a56fe6.png)

Hovering the inserter button makes it lighter. Hovering the Publish button makes it darker.
![image](https://user-images.githubusercontent.com/19592990/73964075-25f1e600-48d7-11ea-96a8-12dc0fcf1ff1.png)
![image](https://user-images.githubusercontent.com/19592990/73964104-330ed500-48d7-11ea-91d3-e381e2f9e0bf.png)

jasmussen commented:
Excellent feedback. I haven't gotten to all of that yet, and it is my intention to do so, and your screenshots are still helpful!

jasmussen commented:
Got a lot of things done today. 

One thing I'd like to surface is https://github.com/WordPress/gutenberg/pull/19348/commits/410befdf226298e8cb67e513fa821d77cc072289, which makes the separator easier to select. I could swear there was an issue for this, but perhaps I'm misremembering?

In any case, it now leverages the block margins to make the thing selectable. Perhaps it's the lack of block borders and block paddings that makes this a non issue, compared to the past. 

Frontend:

<img width="763" alt="front" src="https://user-images.githubusercontent.com/1204802/74028621-180d9500-49ab-11ea-8b5c-cf19a5b1359c.png">

Editor:

<img width="778" alt="editor" src="https://user-images.githubusercontent.com/1204802/74028625-1b088580-49ab-11ea-8918-71daea30fd65.png">

jasmussen commented:
Polished top toolbar and simplified the CSS a bit. The mover control is now the same.

![top toolbar](https://user-images.githubusercontent.com/1204802/74143116-7d0afa00-4bfa-11ea-9ef2-15191471821a.gif)

Works horizontally as well of course:

<img width="464" alt="Screenshot 2020-02-10 at 11 42 33" src="https://user-images.githubusercontent.com/1204802/74143158-8c8a4300-4bfa-11ea-9467-0828ac8e92b3.png">

jasmussen commented:
In https://github.com/WordPress/gutenberg/pull/19348/commits/8006ab0158c77288a6a2ee6766fca7734cde1a7d I polished the placeholder component.

I adjusted a font size, some paddings, and removed the icon. This G2 effort overall removes quite a few icons in many places, including dropdown menus. This is especially useful when placeholders are shown in smaller contexts, but brings a bit consistency across those dimensions:

<img width="865" alt="Screenshot 2020-02-10 at 12 50 30" src="https://user-images.githubusercontent.com/1204802/74148035-c5c7b080-4c04-11ea-8d9f-49eaed53665b.png">

Note, those hanging plus icons are not unique to this branch, they are showing because of a rule (which we should revisit) of showing the appender after any non-text block. Even when the block is deselected. It's not a good rule.

jasmussen commented:
I polished menus and the block switcher ever so slightly, and I discovered this:

![popover direction](https://user-images.githubusercontent.com/1204802/74150201-b4cd6e00-4c09-11ea-9982-454af6133126.gif)

Popovers open in lots of different direction. Why? I understand there's some math going on to make sure it opens in a direction that has sufficient space for it to be visible, i.e. don't open to the right if you're close to the right edge of the screen. But in the GIF example above, there's room in the same direction for all the menus, so why don't they all open from top left and in the direction of down and right?

ZebulanStanphill commented:
@jasmussen

The "Add block" button is the same color as the focus border, and there's no separation between the two, so you can't immediately tell if it is focused or not.

I agree that the label and description could be merged to say something simple like "Insert image" or "Insert media" (in the case of the Media & Text block). Also, I think it is a bit inconsistent to have one button labeled "Media Library" and another labeled "Insert from URL"; shouldn't they both start (or not start) with "Insert from"?

jasmussen commented:
Great thoughts, good catches. Lots to do, I'll get there.

jasmussen commented:
In https://github.com/WordPress/gutenberg/pull/19348/commits/0a74609fc281dbb4d0f327885affd33bd9b428e8, I started work on the Button component refactor:

![buttons](https://user-images.githubusercontent.com/1204802/74234115-6f25a980-4ccc-11ea-8e23-8abc6a86f945.gif)

This will need more work, but it's a great way to test the ideas outlined in #18667 in actual practice. I initially thought it was going to be an issue to remove the border around the secondary button, but in practice so far, it feels like the value that the fewer borders brings to clarity and hierarchy has substantial value, and the context makes it obvious what's button, just like how buttons in a dropdown menu still obviously remain dropdown menu items. 

We'll need more work here, on margins, paddings, font weight and heuristics. But it's a good time to test so far.

ZebulanStanphill commented:
Things are really feeling a lot more lightweight compared to `master`. :+1:

Some more observations as of the latest changes:

The inline text color indicator doesn't quite work with the current toolbar button focus style.
![image](https://user-images.githubusercontent.com/19592990/74253295-835dac80-4cb4-11ea-9266-ecfcbba97d32.png)

The revisions link seems to be lacking some margin/padding.
![image](https://user-images.githubusercontent.com/19592990/74253834-3928fb00-4cb5-11ea-8013-bf7ef707f5f0.png)

The block style/transform popover has a bunch of padding now. Is that intentional?
![image](https://user-images.githubusercontent.com/19592990/74255462-8908c180-4cb7-11ea-818c-a51a0e966c6b.png)

The Save/switch to draft button changes text color slightly on hover, but the Preview button does not.
![image](https://user-images.githubusercontent.com/19592990/74253446-b56f0e80-4cb4-11ea-9bf4-fed22c7509c0.png)

The subtle background when you press the Preview button is not present for the Save/switch to draft button. Icon-only buttons also lack this style.
![image](https://user-images.githubusercontent.com/19592990/74253891-4e058e80-4cb5-11ea-89e8-b584b280a7c4.png)

Additionally, it seems a bit odd that this style is being used for a pressed state for secondary text buttons, but elsewhere it is used as a hover style. That seems inconsistent to me.
![image](https://user-images.githubusercontent.com/19592990/74254124-a2a90980-4cb5-11ea-95a0-f387e7c28ef7.png)
![image](https://user-images.githubusercontent.com/19592990/74254190-b9e7f700-4cb5-11ea-89c0-e1ca25652918.png)

Meanwhile, the Block navigation menu uses it to show the currently-selected block, which aligns more with the current usage for the secondary text buttons.
![image](https://user-images.githubusercontent.com/19592990/74254418-09c6be00-4cb6-11ea-9a54-71e2f552fd87.png)

I'm not sure how to solve this, but one big issue here is that blue is used as the foreground/text color for icon buttons, but text-only buttons use it as the default text color, so you can't use the same hover style for text buttons and icon buttons.

The answer might be to make text buttons use a black text color (just like dropdown menus), but I worry that might make it too hard to distinguish what is a button and what is just text. What do you think? It's worth noting that the "Replace" button in the block toolbar is already black.

The opposite direction, making all icon buttons blue by default, seems like it would mess with the contrast too much, so I think that's out of the question.

jasmussen commented:
Incredible feedback as usual. I'm out for the day but rest assured I'll work through it!

Commit:
Try: Briefly flash footprint on first click.

Commit:
Adjust primary, secondary and tertiary buttons.

This will no doubt need additional feedback, and the branch should stay open for a while to gather all those thoughts.

One of the primary reasons for reducing the borders is, similar to tertiary buttons, that they work due to their context, just like how menu ite
ms work in their menu item contexts. This puts a great onus on the implementer to choose the right button for the right context. In that vein, w
e may even want to provide additional buttons, such as a solid background button that isn't primary, for cases where a primary button is insuffi
cient, and a secondary button does not work well enough in the context. An example could be considered the link dialog mocked up here https://github.com/WordPress/gutenberg/issues/20007#issuecomment-581812491

jasmussen commented:
I pushed some fixes, including to the text color, polish to the transform dialog, and better code for the buttons. 

Your point on the hover vs. selected states is well made. I suspect the buttons need a little more attention, and certainly feedback. Visuals speak better so here are a couple of GIFs:

![icons](https://user-images.githubusercontent.com/1204802/74316985-d8fc8c80-4d7a-11ea-8cf3-03e1e54c48f9.gif)

![header](https://user-images.githubusercontent.com/1204802/74316987-da2db980-4d7a-11ea-8832-6470773fbb4c.gif)

![nav](https://user-images.githubusercontent.com/1204802/74316991-dbf77d00-4d7a-11ea-8b02-d6bba0ecd9e8.gif)

![toggled](https://user-images.githubusercontent.com/1204802/74317106-1103cf80-4d7b-11ea-9942-d121775b9554.gif)

![switcher](https://user-images.githubusercontent.com/1204802/74316994-ddc14080-4d7a-11ea-9865-3f7c0bed5bda.gif)

Of those, to me the heuristics feel sensible (if able to receive more polish) with regards to hover, active and focus styles. The total inversion for toggled and selected states feels very powerful in every situation except perhaps the style picker, where it feels too much. I wonder if that's a matter of unifying the border radius or overall looking at a different treatment. But for now it felt important to try it out in principle.

jasmussen commented:
Polished the gallery:

![gallery](https://user-images.githubusercontent.com/1204802/74319108-b2405500-4d7e-11ea-866a-4cbbd441b2fc.gif)

This block deserves a refactor, separately. Notably we had this idea:

<img width="744" alt="Screenshot 2020-02-12 at 10 01 27" src="https://user-images.githubusercontent.com/1204802/74319130-b9676300-4d7e-11ea-97c5-b029032306dc.png">

Increasingly this seems like a less good idea, because it separates the gallery from the caption below. It seems like with the "Replace" button landing on Images, a similar one could be added to the gallery.

jasmussen commented:
Alright — I finished that last item on my list. There's a lot still to do, but some of that I've so far cautiously put on a list called "followup". That list includes some non-trivial things such as the sidebar and other aspects, which obviously need addressing. However I'm trying to find a balance here between keeping the already-substantial change this G2 effort is at a place where _enough is addressed that a proper feel for the change can be had_, while at the same time keeping the pull requests from growing any bigger. 

My hope is that this balance lets us act on feedback this branch can receive in the coming weeks (and for as long as necessary), and hopefully result in proving the overall concept deserving of being merged into the master branch not too long after the 5.4 cycle has ended. Provided that is possible, it should give us ample time to address the items so far unaddressed.

ZebulanStanphill commented:
I think this branch is already at a really good state. I think a lot of the remaining issues can be handled with later. There are a few things I would like to see addressed before a merge, though:

When clicking a block, a light gray overlay appears briefly before fading away. Is that intentional?

Are we sure we want to get rid of the accordions in the block style/transform switcher?

The block toolbar ellipsis dropdown is positioned a bit too far down when it opens downwards. It's fine when opening upwards, though. Also, the Remove from Reusable blocks button still needs to have its icon removed.
![image](https://user-images.githubusercontent.com/19592990/74353749-fa13ac00-4d7f-11ea-85e7-3b7a1357db94.png)

jasmussen commented:
I had meant to polish a bit further today, but landed massive cold so I'll have to stay bedridden for the day.

> When clicking a block, a light gray overlay appears briefly before fading away. Is that intentional?

This is an experiment, yes, to briefly indicate the footprint of the block, try and find a balance between _this is an editor that doesn't need borders around everything_, and addressing the specific issue of the toolbar sitting far on the left of a right-aligned short paragraph. It's one of those things that is best tested in the branch, I believe.

> Are we sure we want to get rid of the accordions in the block style/transform switcher?

Eventually, probably. For this v1 branch? Possibly not? In removing them I had hoped to take the first step towards this idea:

<img width="1440" alt="Drop downs" src="https://user-images.githubusercontent.com/1204802/74407525-76ff5e00-4e32-11ea-8587-2cdda05b16c8.png">

That idea is one of reducing the overall footprint of the switcher and simplifying, while still hopefully making styles more useful. It will no doubt require more dev work and iteration separately, but there's going to be a phase where we're in-between. Open to suggestions here. 

> The block toolbar ellipsis dropdown is positioned a bit too far down when it opens downwards. It's fine when opening upwards, though.

I have some frustrations with how this is currently working, as was noted in https://github.com/WordPress/gutenberg/pull/19348#issuecomment-584100911. 

Interestingly, this is one of those discrepancies that became visible in this branch, but I never thought about it in master — I feel that's a good sign, we've spent a lot of time fixing bugs and growing the featureset and it organically grew into what's shipping. But when giving an intentional focus to the visuals, suddenly a lot of inconsistencies and cruft appear.

I'm not sure how much dev work is required to get the popovers to open consistently — as shown in the switcher screen above, I'd like to try the interface where popovers from the block toolbar are shown in the same visual treatment and with the same spacing between elements. And when there's space, it feels like the dropdowns should all open in the same direction. I'll see if I can make the spacing consistent, at least.

ZebulanStanphill commented:
> I had meant to polish a bit further today, but landed massive cold so I'll have to stay bedridden for the day.

Hope you get well soon! I just recently got over being sick myself.

> > When clicking a block, a light gray overlay appears briefly before fading away. Is that intentional?
> 
> This is an experiment, yes, to briefly indicate the footprint of the block, try and find a balance between _this is an editor that doesn't need borders around everything_, and addressing the specific issue of the toolbar sitting far on the left of a right-aligned short paragraph. It's one of those things that is best tested in the branch, I believe.

Well, it seems to work alright. I'm not really sure how effective it is, but it doesn't seem to have any negative effects as far as I can tell.

> > Are we sure we want to get rid of the accordions in the block style/transform switcher?
> 
> Eventually, probably. For this v1 branch? Possibly not? In removing them I had hoped to take the first step towards this idea:
> 
> <img alt="Drop downs" width="1440" src="https://user-images.githubusercontent.com/1204802/74407525-76ff5e00-4e32-11ea-8587-2cdda05b16c8.png">

I wonder how those mockups would work on mobile? Sideways-opening menus tend to have issues on mobile devices, while accordions work fine on all platforms.

jasmussen commented:
Rebased the try/g2 branch, reset this branch to that, cherry picked, verified it ran, then squashed this to make future cherrypicks easier. Hopefully everything works as it should still, it seems to for me :D 

I'll now proceed to adress all the great feedback that's been piling up.

jasmussen commented:
> I wonder how those mockups would work on mobile? Sideways-opening menus tend to have issues on mobile devices, while accordions work fine on all platforms.

I think this is good feedback, and something you should consider sharing on the main ticket itself: #18667. I can think of a number of ways to address that, including simply behaving as a sequence of dialogs (like when you click a dropdown on your phone and it opens a dialog). But it's both something that deserves more thought, and will not be addressed immediately in this PR, so totally worth surfacing.

jasmussen commented:
This PR fixes #20254.

jasmussen committed:
Polish preview dropdown.

jasmussen committed:
Address feedback: Remove icon from menu, and realign menu.

jasmussen committed:
Try: tertiary buttons for media placeholder, with tweaks.

jasmussen committed:
Polish horizontal nav and text footprint experiment.

This moves the flashing footprint indicator to only the paragraph block.

youknowriad commented:
<img width="239" alt="Capture d’écran 2020-02-19 à 11 36 21 AM" src="https://user-images.githubusercontent.com/272444/74826550-124d7300-530c-11ea-8eb8-e18d3beb8387.png">

I tried updating the button height to 36px to keep it consistent with the other buttons in the top bar. It looks way better for me. Also, I learned that everytime we create a specific inconsistency, we kind of go back later, so my preference would be to stay consistent here.

It doesn't work that well for the sibling inserter but I believe the sibling inserter doesn't work at all as a "primary" button (blue).

jasmussen commented:
Sibling inserter in 36x36 for reference:

<img width="748" alt="Screenshot 2020-02-19 at 11 40 37" src="https://user-images.githubusercontent.com/1204802/74826918-b505f180-530c-11ea-8109-6159cb021c6f.png">

youknowriad commented:
I noticed that the panel header (block | document) in the sidebar is now white, I think ultimately that's a good change but right now, because the sub panels (inspector panels and document settings) share the same design (white with light gray border), I feel it removes some hierarchy. So I wonder if we should do it now or only when we revisit the design of these panels.

jasmussen commented:
White sidebar tabs, for reference:

<img width="454" alt="Screenshot 2020-02-19 at 15 14 36" src="https://user-images.githubusercontent.com/1204802/74842433-b514ea00-532a-11ea-8969-4080952a6280.png">

Happy to address based on any consensus, and I'm eager to see if we can merge this into G2 soon and be that much closer 🤞

jasmussen commented:
I rebased. Here's a little taste test to whet the appetites for a merge into g2:

![g3](https://user-images.githubusercontent.com/1204802/74842915-66b41b00-532b-11ea-9803-996a46646cd3.gif)

I noticed that when the Top Toolbar option is enabled, the movers still do the smart unhiding. We should probably just toggle that off up there. @ItsJonQ can you help with that? Can commit straight to the try/g2 branch now :)

ItsJonQ commented:
@jasmussen Taking a look when I can :D

ItsJonQ commented:
@jasmussen Haiii! The Movers seem to be behaving on my end :o. I've been resizing desktop <-> mobile, as well as loading the page at various viewport sizes.

In doing so, I discovered this:

<img width="778" alt="Screen Shot 2020-02-19 at 9 48 00 AM" src="https://user-images.githubusercontent.com/2322354/74845517-23db4e80-52fd-11ea-919a-a5f0d70ecc76.png">

Not sure if this already exists. It's when the viewport is **exactly** 782px

jasmussen commented:
Documenting to fix either in this branch or in g2 or in a followup: the block toolbar when seen in one of the responsive views (from the new dropdown) becomes a broken mix of what it is in the top toolbar and a by-the-block version.

jasmussen commented:
Q can you clarify what the issue is in that screenshot, it's hard to tell?

Also, what I mean about the top toolbar is when it's permanently on:

<img width="383" alt="Screenshot 2020-02-19 at 15 53 25" src="https://user-images.githubusercontent.com/1204802/74845945-007dc700-5330-11ea-90cc-d55270f50007.png">

In that case the movers keep their appear behavior:

<img width="1448" alt="Screenshot 2020-02-19 at 15 53 32" src="https://user-images.githubusercontent.com/1204802/74845961-05427b00-5330-11ea-827e-30b6cee6fc92.png">

It seems to work fine on the mobile breakpoints where the toolbar is forced up there.

Commit:
Add mover support for fixed top toolbar

ZebulanStanphill commented:
@youknowriad
> It doesn't work that well for the sibling inserter but I believe the sibling inserter doesn't work at all as a "primary" button (blue) because it's not a primary action anyway.

I agree; the sibling inserter shouldn't be considered a primary button.

Commit:
Address feedback.

- Restore gray sidebar header.
- Use isPrimary for toolbar library button.
- Make inbetweenserter not look primary.

jasmussen commented:
Good thoughts. 

I think many of these thoughts are based on the half-step, even if it's still a big step, that this branch is. The sidebar needs a LOT of love, and a white header takes a step towards that. But I'm fine to explore that in a followup, so it's now gray again, even if that looks out of place with the reduced color set — (most of the rest of the sidebar also looks out of place :D ):

<img width="453" alt="Screenshot 2020-02-20 at 09 48 14" src="https://user-images.githubusercontent.com/1204802/74916576-64eb6580-53c6-11ea-81b7-c4173cac917d.png">

I also made the block library button in the toolbar into an isPrimary button, which allowed me to remove a bunch of CSS. I kept it 32x32 as it's part of aligning those buttons to a 12px grid. I'm happy to also revisit this in a followup and either change that to 36, or explore the 32 in other places.

By making the inserter a primary, ironically, it also made the sibling inserter primary, which meant I had to undo some of those visuals to make it appear not primary:

<img width="202" alt="Screenshot 2020-02-20 at 09 48 20" src="https://user-images.githubusercontent.com/1204802/74916743-aed44b80-53c6-11ea-9807-997b25b8d71e.png">

I don't have strong feelings for the current state of the sibling inserter, I profoundly thing that inserters inside the canvas deserve a whole iteration on its own — can we remove a bunch of them, for example? On the block library ticket that [discusses opening the block library as a modal dialog that defaults to the "Patterns" tab](https://github.com/WordPress/gutenberg/issues/17335#issuecomment-562049752), we made it blue, like so:

<img width="1202" alt="Basic Editor" src="https://user-images.githubusercontent.com/1204802/74916814-d0cdce00-53c6-11ea-9fb4-c426a7198229.png">

But given it needs more love, and it's holding back this PR, absolutely something to follow up on.

youknowriad commented:
Thanks for the updates @jasmussen that's great

> By making the inserter a primary, ironically, it also made the sibling inserter primary, which meant I had to undo some of those visuals to make it appear not primary:

I believe we need to have a prop in the inserter to choose whether we want a primary button or not to avoid having to revert the changes in CSS. For example in appenders... I'll see if I can make this change.

youknowriad committed:
Fix inserter toggle in different situations

youknowriad commented:
Now, that this branch is very close to G2 and without touching the panels... I think we can merge it with G2 and continue on a single branch. Nice work @jasmussen

jasmussen commented:
ƪ(˘⌣˘)┐ ƪ(˘⌣˘)ʃ ┌(˘⌣˘)ʃ
